src/main/scala/com/databricks/spark/xml/functions.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.xml
+
+import org.apache.spark.annotation.Experimental
+import org.apache.spark.sql.{Column, Dataset, Row}
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
+import org.apache.spark.sql.types.{ArrayType, DataType, StructType}
+
+import com.databricks.spark.xml.parsers.StaxXmlParser
+import com.databricks.spark.xml.util.InferSchema
+
+/**
+ * Support functions for working with XML columns directly.
+ */
+object functions {
+
+  /**
+   * Infers the schema of XML documents as strings.
+   *
+   * @param ds Dataset of XML strings
+   * @param options additional XML parsing options
+   * @return inferred schema for XML
+   */
+  @Experimental
+  def schema_of_xml(ds: Dataset[String], options: Map[String, String] = Map.empty): StructType =
```

Comment:
Sorry for a bit of forth and back. I believe `snake_case` naming was introduced fro compatibility of SQL as of SPARK-10621. I think `schema_of_xml` and `from_xml_string` are actually not the SQL-like functions as it takes whole datasets or take other types of signature.

Maybe it's better to move both to `https://github.com/databricks/spark-xml/blob/master/src/main/scala/com/databricks/spark/xml/package.scala` as proposed before just, as static functions (or as an implicit against `Dataset[String]` if applicable).

Reply:
So move all of functions.scala to package.scala? and rename to something like inferSchema again?

Reply:
I meant to move, from `functions.scala` to `package.scala` with renaming, only functions that don't have `Column` signature because the functions with `Column` signature as input/output are considered as SQL functions. Namely, we might have to move `schema_of_xml`, `schema_of_xml_array` and `from_xml_string`.